packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/helpers/migration.ts

```diff
@@ -39,4 +53,41 @@ export class MigrationHelper {
   migrateStorageDocument(document: SavedObjectUnsanitizedDoc): SavedObjectUnsanitizedDoc {
     return this.migrator.migrateDocument(document, { allowDowngrade: true });
   }
+
+  async migrateAndDecryptStorageDocument<T, A extends string>({
+    document,
+    typeMap,
+    originalAttributes,
+  }: {
+    document: SavedObjectUnsanitizedDoc<T> | SavedObject<T>;
+    typeMap: AuthorizationTypeMap<A> | undefined;
+    originalAttributes?: T;
+  }): Promise<SavedObject<T>> {
```

Comment:
The core addition of the PR, `migrateAndDecryptStorageDocument` will execute the `migrate` and `decrypt` "steps" in order, depending on if the document needs to be downgraded (eviction schema) or upgraded (migration)